subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java

```diff
@@ -572,13 +570,9 @@ private boolean resolveFromDefaultImports(ClassNode type, boolean testDefaultImp
                     return true;
                 }
             }
-            for (String gradlePublicPackage : gradlePublicPackages) {
-                if (resolveFromResolver(type, gradlePublicPackage + "." + name)) {
-                    if ("org.gradle.util".equals(gradlePublicPackage)) {
-                        deprecatedImports.add(name);
-                    }
-                    return true;
-                }
+            if (resolveFromResolver(type, "org.gradle.util" + "." + name)) {
```

Comment:
Looks like this changes the original behavior. `for (String gradlePublicPackage : gradlePublicPackages)` loop was trying to resolve a symbol from default packages. Without this, lots of default imports couldn't be resolved (e.g. `org.gradle.api.*`).